src/fight.cpp

```diff
@@ -1933,7 +1933,7 @@ int weapon_special(ObjData *wpn, CharData *ch) {
 }
 
 void hit(CharData *ch, CharData *victim, int type) {
-    int victim_ac, calc_thaco, dam, diceroll, weapon_position, hidden;
+    int victim_ac, calc_thaco, dam, diceroll, weapon_position, hidden, hit;
```

Comment:
Please rename this new `hit` variable to `to_hit` or something so it doesn't shadow the function name.
Also, please make it a boolean. (`bool`)